src/HealthCheck.Core/CheckerBase.cs

```diff
@@ -1,26 +1,42 @@
 ﻿using System;
 using System.Threading.Tasks;
-using HealthCheck.Core.Helpers;
+using HealthCheck.Core.Extensions;
 
 namespace HealthCheck.Core
 {
     public abstract class CheckerBase : IChecker
     {
-        private TimeSpan? _timeout = TimeSpan.FromSeconds(5);
+        public string Name { get; set; }
+        public string SectionName { get; set; }
+        public virtual bool PreserveContext => false;
+        public virtual TimeSpan? Timeout { private get; set; }
```

Comment:
Public setter and private getter seems a bit strange, let's make the getter public as well.